### PR TITLE
Mardizzone/pos 493

### PR DESCRIPTION
**Pull request by marcello33**
This PR provides benchmark tests for the top 5 time consuming functions involved in the `stateSynced` events flow.
Thanks @JekaMas for the tips ;) 

```goos: darwin
goarch: amd64
pkg: github.com/maticnetwork/heimdall/bridge/setu/processor
cpu: Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz


BenchmarkSendStateSyncedToHeimdall-12
    4644	    270211 ns/op	   49095 B/op	     497 allocs/op

BenchmarkIsOldTx-12
    21813	     56133 ns/op	   10829 B/op	      70 allocs/op

BenchmarkSendTaskWithDelay-12
    5964	    252384 ns/op	    6941 B/op	      97 allocs/op

BenchmarkCalculateTaskDelay-12
    4455	    372954 ns/op	   87256 B/op	     473 allocs/op

BenchmarkGetUnconfirmedTxnCount-12
    24110	     47444 ns/op	    8168 B/op	      54 allocs/op

### CODE REVIEW

File: auth/types/account_retriever.go [hunk 1]
```diff
@@ -2,18 +2,28 @@ package types
 
 import (
 	"fmt"
+	cliContext "github.com/cosmos/cosmos-sdk/client/context"
 
 	"github.com/maticnetwork/heimdall/types"
 )
 
+var (
+	NQuerier NodeQuerier
```
Comment (JekaMas) [inline auth/types/account_retriever.go:11]: can we do something to not to have global vatiable?

File: helper/util.go [hunk 2]
```diff
@@ -41,6 +41,15 @@ import (
 	"github.com/maticnetwork/heimdall/types/rest"
 )
 
+//go:generate mockgen -destination=./mocks/http_client_mock.go -package=mocks . HTTPClient
+type HTTPClient interface {
+	Get(string) (resp *http.Response, err error)
+}
+
+var (
+	Client HTTPClient
```
[the remark this answers is not in the document]
Comment (marcello33) [reply helper/util.go:50]: I am afraid we can't, as we want this variable to be a mockHttpClient in tests and a pure rest http client with normal execution

File: auth/types/account_retriever.go (hunk 1 above)
Comment (marcello33) [reply auth/types/account_retriever.go:11]: I am afraid we can't, as we want this variable to be a mockNodeQuerier in tests and a pure &cliContext.CLIContext{} with normal execution

**Review by JekaMas — DISMISSED** [review]

File: auth/types/account_retriever.go [hunk 3]
```diff
@@ -2,18 +2,28 @@ package types
 
 import (
 	"fmt"
+	cliContext "github.com/cosmos/cosmos-sdk/client/context"
```
Comment (begmaroman) [inline auth/types/account_retriever.go:5]: Could you add an empty line above to separate third-party imports from the built-in ones?

**Comment by begmaroman** [comment]
@marcello33 we don't need to duplicate logic. I've separated the function that handles events to be reusable.

File: bridge/setu/listener/rootchain.go [hunk 4]
```diff
@@ -186,8 +186,101 @@ func (rl *RootChainListener) queryAndBroadcastEvents(rootchainContext *RootChain
 			if selectedEvent == nil {
 				continue
 			}
-
-			rl.handleLog(vLog, selectedEvent, pubkeyBytes)
+			logBytes, _ := json.Marshal(vLog)
+			if selectedEvent != nil {
```
[the remark this answers is not in the document]
Comment (marcello33) [reply bridge/setu/listener/rootchain.go:190]: Hey, thanks for the comment 
This is due to conflicts with develop env, I haven't touched that function

File: auth/types/account_retriever.go (hunk 3 above)
Comment (marcello33) [reply auth/types/account_retriever.go:5]: Will do 👌

File: bridge/setu/listener/rootchain.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (marcello33) [reply bridge/setu/listener/rootchain.go:190]: This is downside of having long standing PRs open. Had to continuously merge develop back and still some conflicts were arises. Will remove the duplicates and push again, thanks

**Review by begmaroman — DISMISSED** [review]
One minor comment, then LGTM

**Review by JekaMas — APPROVED** [review]

**Review by begmaroman — APPROVED** [review]

**Pull request merged**